terraform/providers/aws/README.md

```diff
@@ -19,189 +19,193 @@ This project will deploy an end to end infrastructure in AWS that includes the b
+
+Set the below environment variables if you'd like to use Packer & Terraform locally
+
+    $ export AWS_ACCESS_KEY_ID=YOUR_AWS_ACCESS_KEY_ID
+    $ export AWS_SECRET_ACCESS_KEY=YOUR_AWS_SECRET_ACCESS_KEY
+    $ export AWS_DEFAULT_REGION=us-east-1
+    $ export ATLAS_USERNAME=YOUR_ATLAS_USERNAME
+    $ export ATLAS_TOKEN=YOUR_ATLAS_TOKEN
+
+### Generate Keys and Certs
+
+- [ ] Generate certs
+  - [ ] `sh gen_cert.sh hashicorpdemo.com site HashiCorp` in [scripts](../../../scripts)
+  - [ ] `sh gen_cert.sh consul vault HashiCorp` in [scripts](../../../scripts)
+    - These will be placed in a folder called `certs` within the [scripts/.](../../../scripts) directory
+- [ ] Generate keys
+  - [ ] `sh gen_key.sh site` or `sh gen_key.sh site ~/.ssh/my-existing-private-key.pem` in [scripts](../../../scripts)
+    - These will be placed in a folder called `keys` within the [scripts/.](../../../scripts) directory
+  - [ ] Copy the `site.pem` file into [terraform/modules/keys/.](../../modules/keys)
+    - This is a temporary workaround until the `key_file` attribute of the `connection` block within `remote-exec` provisioners can accept file contents instead of just file paths, for the time being this key will be checked into GitHub
+- [ ] Move all keys & certs created here out of the repo and to a secure location
+  - Aside from the workaround mentioned above, no keys or certs should be checked into version control
+
+### Create and Configure Artifacts with Packer in Atlas
+
+Use the [New Build Configuration](https://atlas.hashicorp.com/builds/new) tool to create your Builds. Leave the **Automatically build on version uploads** box unchecked.
+
+After creating each build configuration, there is some additional configuration you'll need to do.
+
+To integrate with GitHub...
+
+- Go into "Integrations" in the left navigation of the Build Configuration
+- Select the `best-practices` GitHub repository you just forked
+- Leave **Packer Directory** blank
+- Enter the **Packer template** provided and click **Associate**
+
+Once configuration is complete you'll want to go to "Builds" in the left navigation and click **Queue build**, this should successfully create a new artifact.
+
+- [ ] Create `aws-ubuntu-base` Artifact
+  - [ ] [Create Build Configuration](https://atlas.hashicorp.com/builds/new): `aws-ubuntu-base`
+  - [ ] In "Integrations": Setup GitHub Integration
+    - **GitHub repository**: `best-practices`
+    - **Packer template**: `packer/aws/ubuntu/base.json`
+  - [ ] Queue new build to create artifact in "Builds"
+
+Wait until the Base Artifact has been created before moving on to the child Build Configurations. These will fail with an error of `* A source_ami must be specified` until the Base Artifact has been created.
+
+For child Build Configurations there is one additional step you need to take. In "Settings", set **Inject artifact ID during build** to `aws-us-east-1-ubuntu-base` for each.
+
+- [ ] Create `aws-us-east-1-ubuntu-consul` Artifact
```

Comment:
Ah I think your comment will be - we need to do this region specific because of the base artifacts. 

Do we think the value that the base artifact provides is greater than the complexity it adds?